src/apply.js

```diff
@@ -1033,7 +1061,12 @@ function processStyle(glStyle, mapOrGroup, styleUrl, options) {
     } else {
       id = glLayer.source || getSourceIdByRef(glLayers, glLayer.ref);
       // this technique assumes gl layers will be in a particular order
-      if (!id || id != glSourceId) {
+      if (
+        // This line is because rasters set properties on their source
+        glLayer.type === 'raster' ||
```

Comment:
This is the line I'm most unsure about. We need this because is we have two layers with the same source they need to apply different operations to their sources.